python/src/triton.cc

```diff
@@ -551,7 +551,7 @@ void init_triton_ir(py::module &&m) {
                return llvm::dyn_cast<mlir::FuncOp>(funcOperation);
              auto loc = self.getUnknownLoc();
              if (auto funcTy = funcType.dyn_cast<mlir::FunctionType>()) {
-               mlir::ArrayRef<mlir::NamedAttribute> attrs = {
+               llvm::SmallVector<mlir::NamedAttribute> attrs = {
```

Comment:
`mlir::NamedAttribute` objects created as part of your initializer for `attrs` are going out of scope once the initializer finishes. That could lead to a segmentation fault when the `ArrayRef` later tries to access them. 

The reason using a `SmallVector` fixes the issue is that the vector will copy the `NamedAttribute` objects internally, so they don't go out of scope.